File: token/token.go
package token

type TokenType int

const (
	_ = iota
	INVALID
	EOF

	IDENTIFIER
	STRING
	INTEGER
	FLOAT

	TRUE
	FALSE
	NULL

	ASSIGN
	PLUS
	MINUS
	STAR
	SLASH
	PLUS_ASSIGN
	MINUS_ASSIGN
	STAR_ASSIGN
	SLASH_ASSIGN

	BIT_AND
	BIT_OR

	NOT
	AND
	OR

	EQUAL
	NOT_EQUAL
	LESS
	GREATER
	GREATER_EQUAL
	LESS_EQUAL

	LEFT_PARENTHESES
	RIGHT_PARENTHESES

	COMMA

	END
	FUNCTION
	CLASS
	EXTENDS

	IF
	ELSE
	WHILE
)

type Token struct {
	Type    TokenType
	Literal string
}

func New(token_type TokenType, literal string) Token {
	return Token{Type: token_type, Literal: literal}
}

File: Clover/Token.cs
using System;

namespace Clover
{
    public enum Token
    {
        Invalid,
        Eof,
        None,
        
        Identifier,
        String,
        Integer,
        Float,
        
        True,
        False,
        Null,
        
        Assign,
        Plus,
        Minus,
        Star,
        Slash,
        PlusAssign,
        MinusAssign,
        StarAssign,
        SlashAssign,
        
        BitAnd,
        BitOr,
        
        Not,
        And,
        Or,
        
        Equal,
        NotEqual,
        Less,
        Greater,
        LessEqual,
        GreaterEqual,
        
        LeftParentheses,
        RightParentheses,
        LeftBracket,
        RightBracket,
        LeftBrace,
        RightBrace,
        
        Comma,
        Colon,
        Dot,
        
        End,
        Local,
        Function,
        Return,
        Class,
        Extends,
        New,
        Base,
        This,
        Constructor,
        At,
        
        If,
        Else,
        While,
        
        Load
    }

    public struct TokenData
    {
        public Token Token;
        public String Value;
        public String Filename;
        public int Line;
    }

}